File: lib/types/pattern.ts
export type PatternStatus = 'draft' | 'published'

export type PatternCategory =
  | 'Architecture'
  | 'Design Patterns'
  | 'AI Prompts'
  | 'Best Practices'
  | 'Code Generation'
  | 'Testing'
  | 'Security'
  | 'Performance'

export type Pattern = {
  id: string
  title: string
  slug: string
  shortDescription: string
  fullContent?: string
  category: PatternCategory
  tags: string[]
  author?: string
  createdDate: string
  updatedDate: string
  voteCount: number
  status: PatternStatus
  isFeatured?: boolean
  isTrending?: boolean
}

export type PatternListItem = Omit<Pattern, 'fullContent'>

File: backend/src/AIEnterprisePatterns.Core/Entities/Pattern.cs
using System.ComponentModel.DataAnnotations;
using AIEnterprisePatterns.Core.Enums;

namespace AIEnterprisePatterns.Core.Entities;

public class Pattern : BaseEntity
{
    [Required, MaxLength(255)]
    public string Title { get; set; } = string.Empty;

    [Required, MaxLength(255)]
    public string Slug { get; set; } = string.Empty;

    [Required]
    public string ShortDescription { get; set; } = string.Empty;

    public string? FullContent { get; set; }

    public PatternCategory Category { get; set; }

    public List<Tag> Tags { get; set; } = new();

    [MaxLength(100)]
    public string? Author { get; set; }

    public DateTime CreatedDate { get; set; }

    public DateTime UpdatedDate { get; set; }

    public int VoteCount { get; set; }

    public PatternStatus Status { get; set; }

    public bool IsFeatured { get; set; }

    public bool IsTrending { get; set; }
}
